### ISSUE TITLE
Add Feature: Rational Quadratic Kernel (Support And Resistance Areas) Algorithm

### ISSUE THREAD

**Issue by summercms** [chartingshow/documentation#279]
## Enhancement idea

- [x] Add Feature: Rational Quadratic Kernel (Support And Resistance Areas) Algorithm.
- [x] Connect to Alerts Modules and add custom alerts, see here: https://github.com/chartingshow/documentation/issues/70
- [x] Connect to Warning Signals module, see here: https://github.com/chartingshow/documentation/issues/267
- [x] Connect to Trading Signals module, see here: https://github.com/chartingshow/documentation/issues/269
- [x] Add Settings.
- [x] Add to data window.
- [x] Update Doc's.
- [x] Add to Themes, see here: https://github.com/chartingshow/documentation/issues/194

## Description

The Rational Quadratic kernel trading algorithm is a type of kernel function used in Gaussian Process Regression (GPR) models. It is derived from the Radial Basis Function (RBF) kernel and can be considered as an infinite sum of RBF kernels with different length-scales.

### Rational Quadratic Kernel Properties

The Rational Quadratic kernel is defined as:

```
k(x, x') = (1 + (x - x')^2 / (2αl^2))^(-α)
```

Where:

- `α` is a scale mixture parameter.
- `l` is the length-scale parameter.

Some key properties of the Rational Quadratic kernel include:

- **Smoothness**: It is an infinitely differentiable function, providing smooth predictions.
- **Long-range Correlations**: It exhibits long-range correlations, meaning data points far apart can still influence each other's predictions.
- **Flexibility**: By varying the `α` parameter, it can approximate the behavior of other kernels like the RBF (`α → ∞`) or the Matérn class of kernels (`α = p/2`, where `p` is a positive integer).

### Support and Resistance Areas

While the Rational Quadratic kernel trading algorithm itself does not directly provide support and resistance areas, the Gaussian Process Regression model using this kernel can be employed to identify such areas based on the predicted mean and uncertainty estimates.

Potential support and resistance areas can be inferred from the predicted mean function, where:

- Local maxima may indicate potential resistance areas.
- Local minima may indicate potential support areas.

Additionally, the uncertainty estimates (e.g., variance) from the GPR model can help quantify the confidence in these potential support/resistance areas. Areas with lower uncertainty are more likely to be significant support/resistance levels.

In summary, the Rational Quadratic kernel trading algorithm offers a flexible and smooth kernel for GPR models, which can then be leveraged to identify potential support and resistance areas in financial data or other time series analysis tasks.

## Links

https://en.wikipedia.org/wiki/Rational_quadratic_covariance_function

https://www.cs.toronto.edu/~duvenaud/cookbook/

**Comment by summercms** [comment]
......Signals for alerts

**Comment by summercms** [comment]
Update `crypto market algo` and `Core algo` codes as well.
